instrumentation/opentelemetry-instrumentation-threading/tests/test_threading.py

```diff
@@ -0,0 +1,232 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+from concurrent.futures import ThreadPoolExecutor
+from typing import List
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.threading import ThreadingInstrumentor
+from opentelemetry.test.test_base import TestBase
+
+
+class TestThreading(TestBase):
+    def setUp(self):
+        super().setUp()
+        self._tracer = self.tracer_provider.get_tracer(__name__)
+        self._mock_span_contexts: List[trace.SpanContext] = []
+        ThreadingInstrumentor().instrument()
+
+    def tearDown(self):
+        ThreadingInstrumentor().uninstrument()
+        super().tearDown()
+
+    def get_root_span(self):
+        return self._tracer.start_as_current_span("rootSpan")
+
+    def test_trace_context_propagation_in_thread(self):
+        self.run_threading_test(threading.Thread(target=self.fake_func))
+
+    def test_trace_context_propagation_in_timer(self):
+        self.run_threading_test(
+            threading.Timer(interval=1, function=self.fake_func)
+        )
+
+    def run_threading_test(self, thread: threading.Thread):
+        with self.get_root_span() as span:
+            expected_span_context = span.get_span_context()
+            thread.start()
+            thread.join()
+
+            # check result
+            self.assertEqual(len(self._mock_span_contexts), 1)
+            self.assertEqual(
+                self._mock_span_contexts[0], expected_span_context
+            )
+
+    def test_trace_context_propagation_in_thread_pool_with_multiple_workers(
+        self,
+    ):
+        max_workers = 10
+        executor = ThreadPoolExecutor(max_workers=max_workers)
+
+        expected_span_contexts: List[trace.SpanContext] = []
+        futures_list = []
+        for num in range(max_workers):
+            with self._tracer.start_as_current_span(f"trace_{num}") as span:
+                expected_span_context = span.get_span_context()
+                expected_span_contexts.append(expected_span_context)
+                future = executor.submit(
+                    self.get_current_span_context_for_test
+                )
+                futures_list.append(future)
+
+        result_span_contexts = [future.result() for future in futures_list]
+
+        # check result
+        self.assertEqual(len(result_span_contexts), max_workers)
+        self.assertEqual(
+            len(result_span_contexts), len(expected_span_contexts)
+        )
+        for index, result_span_context in enumerate(result_span_contexts):
+            self.assertEqual(
+                result_span_context, expected_span_contexts[index]
+            )
```

Comment:
Is something missing if we just compare them directly?

```suggestion
        self.assertEqual(result_span_contexts, expected_span_contexts)
```

Reply:
You are right. I made a mistake. I think the Python like other languages such Go or Java, if we compared array or object, they will compare the address. I have updated it, thanks.